### PR TITLE
change oc adm mirror in order to use oc-mirror

### PR DESCRIPTION
**Pull request by alknopfler**
Signed-off-by: Alberto Morgante Medina <alknopfler@users.noreply.github.com>

# Description

This is quite a big change and it did a refactor on several areas of the
code base. The gist and most important parts of this change are:

- The ocp-mirror and olm-mirror have now been merged
- The mirror is now done using `oc-mirror`
- The mirror step only syncs the required packages from selected
channels instead of downloading the latest from every channel as it was
doing before. Architectures are not filtered.
- There is now a single organization in Quay called `ztpfw`
- ICSPs and catalogsources are created by oc-mirror
- The catalogsource names are kept as the upstream ones to avoid having
to mess with the subscriptions.

Minor changes:
- Several sleeps have been removed in favor of waiting for resources

## Type of change

Please select the appropriate options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This change is a documentation update

## Testing

It was mostly manually tested. Here's the output of the execution:

This was a retry so the edgecluster deployment was just a veriy. Focus on the olm sync step. Note that the OCP step is gone, it's now all managed by OLM:

image

**Test Configuration**:

- Versions:
- Hardware:

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [ ] A change should not be merged unless it passes CI or there is a comment/update saying what testing was passed.
- [ ] PRs should not be merged unless positively reviewed.

### CODE REVIEW

**Review by alknopfler — COMMENTED** [review]
LGTM

File: deploy-disconnected-registry/common.sh [hunk 1]
```diff
@@ -138,6 +137,35 @@ if [[ $# -lt 1 ]]; then
...
+    echo ">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"
+    echo "creating config for: ${1}"
+    cat <<EOF >${OUTPUTDIR}/oc-mirror-config-${1}.yaml
+apiVersion: mirror.openshift.io/v1alpha2
+kind: ImageSetConfiguration
+storageConfig:
+  registry:
+    imageURL: $DESTINATION_REGISTRY/ocp4/oc-mirror
+mirror:
+  operators: # Operators we want to mirrorEOF
```
Comment (achuzhoy) [inline deploy-disconnected-registry/common.sh:151]: is the EOF in this line needed?

File: deploy-disconnected-registry/olm-sync.sh [hunk 2]
```diff
@@ -74,11 +108,16 @@ function mirror() {
...
+
     [ -d ~/.docker ] || mkdir  ~/.docker
+    sleep 15
+    echo ">>>> Copy pull secret to ~/.docker/config.json"
+    mkdir -p ~/.docker
     cp -f ${PULL_SECRET} ~/.docker/config.json
-
     mkdir -p /var/run/user/0/containers
     cp -f ${OUTPUTDIR}/pull-secret.json /var/run/user/0/containers/auth.json
+    cp -f ${PULL_SECRET} /var/run/user/0/containers/auth.json
```
Comment (achuzhoy) [inline deploy-disconnected-registry/olm-sync.sh:119]: This overwrites the previous copy making it redundant.

File: deploy-disconnected-registry/common.sh (hunk 1 above)
Comment (flaper87) [reply deploy-disconnected-registry/common.sh:151]: EOF is not needed, leftover. Will clean

File: deploy-disconnected-registry/olm-sync.sh (hunk 2 above)
Comment (flaper87) [reply deploy-disconnected-registry/olm-sync.sh:119]: Love that you noticed our "desperate double copy" line. :D

**Review by flaper87 — APPROVED** [review]
waiting for @achuzhoy sanity check/test

**Review by flaper87 — COMMENTED** [review]
/hold

something is still not right with this PR

**Pull request merged**